### DISCUSSION TITLE
Roadmap/project status page

### DISCUSSION THREAD

**Discussion by legezam** [grain-lang/grain#1617] [Ideas]
Hey

I am currently looking for a language to implement a project that is meant to target WASM. I was browsing across the entire WASM landscape, found many appealing languages to use, and came across Grain. Syntactically, Grain is the one that is closest to my heart, however, I am not able to assess the current state of the language. Ie what is supported, what is not, what's ahead, what will never be. Kind of roadmap/feature list. Something like what AssemblyScript has:

https://www.assemblyscript.org/status.html

It would help people greatly to decide quicly whether Grain is an option or not for them. For instance, AssemblyScript states unions are not supported and won't therefore it's a clear no-go for me.

In case you already have such a thing, then I am sorry, but the fact that I wasn't able to find it within 10mins of dedicated search might tell something even about its place :)

What is your opinion on that?

Thanks!

**Comment by ospencer** [comment]
We've talked about having something like this for a while, but it hasn't happened. We've got a big website redesign coming soon™️ and I'll make sure we've got a page like that.

**Reply by legezam** [reply]
@ospencer , thanks for your reply.

Meanwhile, is there a discussion/forum topic, anything not formal anywhere which would give a rough/vague picture of the current status of the project? :)

**Reply by ospencer** [reply]
There isn't really an all-in-one place, but the bulk of the conversation happens on [our Discord server](https://discord.com/invite/grain-lang). You can ask any questions you have in #support; we're pretty quick to respond 🙂 

If you're mostly asking about what WebAssembly features are used, that's an easy one, but as you can imagine the roadmap is a bit fuzzier, but we can speak to it generally.

**Reply by legezam** [reply]
Thank you so much!

**Comment by ospencer** [comment]
Tracking issue here: https://github.com/grain-lang/grain-lang.org/issues/345

**Comment by MaxGraey** [comment]
> For instance, AssemblyScript states unions are not supported and won't therefore it's a clear no-go for me.

May I ask where this is stated? Unions will be supported in the future

**Reply by legezam** [reply]
![image](https://user-images.githubusercontent.com/3974359/214322707-52e8b038-f1d8-4e0e-8529-b362e928c196.png)

Indeed it doesn't explicitly say it won't be supported, but to me "they are not supported by design" says it is something we don't want. Am I wrong?

**Comment by MaxGraey** [comment]
> ![image](https://user-images.githubusercontent.com/3974359/214322707-52e8b038-f1d8-4e0e-8529-b362e928c196.png)
> 
> Indeed it doesn't explicitly say it won't be supported, but to me "they are not supported by design" says it is something we don't want. Am I wrong?

Thanks! [It will be edited](https://github.com/AssemblyScript/website/pull/192). Currently union types are not supported (except for nullable), but that does not mean that this will not change in the future.
